src/record/delete/validator/index.ts

```diff
@@ -0,0 +1,57 @@
+import type { KintoneRestAPIClient } from "@kintone/rest-api-client";
+import type { RecordNumber } from "../types/field";
+import {
+  isValidRecordNumber,
+  hasAppCode,
+  getAllRecordIds,
+  getRecordIdFromRecordNumber,
+} from "../usecases/delete/record";
+
+import { ErrorHandler } from "./errorHandler";
+import { ValidatorError } from "./error";
+
+export const validateRecordNumbers: (
+  apiClient: KintoneRestAPIClient,
+  app: string,
+  appCode: string,
+  recordNumbers: RecordNumber[]
+) => Promise<void> = async (apiClient, app, appCode, recordNumbers) => {
+  const validValues: string[] = [];
+  const errorHandler = new ErrorHandler();
+  const hasAppCodePrevious = hasAppCode(recordNumbers[0].value, appCode);
+  const allRecordIds = await getAllRecordIds(apiClient, app);
+  recordNumbers.forEach((recordNumber, index) => {
+    const value = recordNumber.value;
+    if (value.length === 0 || !isValidRecordNumber(value, appCode)) {
+      errorHandler.addInvalidValueError(recordNumber);
+      return;
+    }
+
+    const _hasAppCode = hasAppCode(value, appCode);
+    if (_hasAppCode !== hasAppCodePrevious) {
+      throw new ValidatorError(
+        "The record number should not be mixed with those with and without app code"
+      );
+    }
+
+    if (validValues.indexOf(value) > -1) {
+      errorHandler.addDuplicatedError(recordNumber);
+      return;
+    }
+
+    if (
+      allRecordIds.indexOf(
+        getRecordIdFromRecordNumber(recordNumber, appCode)
+      ) === -1
+    ) {
+      errorHandler.addNotExistsError(recordNumber);
+      return;
+    }
+
+    validValues.push(value);
+  });
+
+  if (errorHandler.hasError()) {
+    errorHandler.throwError();
```

Comment:
IMO: It is better to generate an Error on `errorHandler` and throw it at here to visualize throwing Error explicitly.

```
throw errorHandler.generateError()
```